File: src/dataset/Java/sieve_exercism-ThomasZumsteg/gemini/GeminiFlash_Sieve_v2.java
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;

public class Sieve {
    private boolean[] sieve;
    private int limit;

    public Sieve(int limit) {
        this.limit = limit;
        sieve = new boolean[limit + 1];
    }

    public List<Integer> getPrimes() {
        if (limit < 2) {
            return new ArrayList<>();
        }

        Arrays.fill(sieve, false);
        List<Integer> primes = new ArrayList<>();
        primes.add(2);

        int sqrtLimit = (int) Math.sqrt(limit);

        for (int i = 3; i <= sqrtLimit; i += 2) {
            if (!sieve[i]) {
                for (int j = i * i; j <= limit; j += 2 * i) {
                    sieve[j] = true;
                }
            }
        }

        for (int i = 3; i <= limit; i += 2) {
            if (!sieve[i]) {
                primes.add(i);
            }
        }

        return primes;
    }
}

File: src/dataset/javascript/sieve_exercism-ThomasZumsteg/gemini/GeminiFlash_sieve_v2.js
var Sieve = function(size) {
    this.primes = [];
    if (size < 2) return;

    const sieve = new Array(size + 1).fill(true);
    sieve[0] = sieve[1] = false;

    for (let p = 2; p <= Math.sqrt(size); p++) {
        if (sieve[p]) {
            for (let i = p * p; i <= size; i += p) {
                sieve[i] = false;
            }
        }
    }

    for (let i = 2; i <= size; i++) {
        if (sieve[i]) {
            this.primes.push(i);
        }
    }
};

module.exports = Sieve;